### PR TITLE
feat: add translation API

### PR DESCRIPTION
**Pull request by h1z3y3**
Close https://github.com/micro/services/issues/187

Guide for getting appkey and secret: https://ai.youdao.com/doc.s#guide

After you sign up as a registered user, you can get some trial balance.

### CODE REVIEW

**Comment by asim** [comment]
We will struggle to use this in English speaking countries. It's more likely we need to use Google translate.

**Comment by h1z3y3** [comment]
Yeah. No problem. It’s another option. I will implement google translations too

**Comment by asim** [comment]
What it does demonstrate is a longer term model where we could support multiple providers, but maybe not yet

**Comment by h1z3y3** [comment]
OK. No problem.
I'll replace it later 😅

**Comment by h1z3y3** [comment]
@asim  I've done with resurfacing translation API by Google translate, please review ;P

**Comment by asim** [comment]
Can you call the "translate" api and then have the endpoint Text. So it will be /v1/translate/Text

File: translation/proto/translation.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  // The translation result
+  string text = 1;
+  // The source of the query string
+  string source = 2;
+  // The model used in translation
+  string model = 3;
+}
+
+// BasicTranslationRequest is the basic edition request
+message BasicTranslationRequest {
```
Comment (asim) [inline translation/proto/translation.proto:22]: Ensure request/response names match rpc name e.g `rpc Text` == `TextRequest` and `TextResponse`

Comment (h1z3y3) [reply translation/proto/translation.proto]: It makes sense.
Resolved

**Pull request merged**